tests/compile-fail/weak_memory/imperfectly_overlapping.rs

```diff
@@ -1,27 +1,28 @@
 // ignore-windows: Concurrency on Windows is not supported yet.
 #![feature(atomic_from_mut)]
+#![feature(core_intrinsics)]
 
+use std::intrinsics::atomic_load;
 use std::sync::atomic::Ordering::*;
 use std::sync::atomic::{AtomicU16, AtomicU32};
 
-// Strictly speaking, atomic accesses that imperfectly overlap with existing
-// atomic objects are UB. Nonetheless we'd like to provide a sane value when
-// the access is not racy.
 fn test_same_thread() {
     let mut qword = AtomicU32::new(42);
     assert_eq!(qword.load(Relaxed), 42);
-    qword.store(u32::to_be(0xabbafafa), Relaxed);
+    qword.store(0xabbafafa, Relaxed);
 
     let qword_mut = qword.get_mut();
 
     let dwords_mut = unsafe { std::mem::transmute::<&mut u32, &mut [u16; 2]>(qword_mut) };
```

Comment:
`transmute` on references is nasty since it infers arbitrary lifetimes.
Could you make this into a separate function with signature `fn(&mut u32) -> &mut [u16; 2]`? That would be a sound function, then this is all safe code -- which also demonstrates that there definitely should not be any UB thrown here.